Add YouTube clips "expando"

thomasasfk opened the pull request:
- Can now expand and watch YouTube clips
- Supports full embedded player functionality

Image of the new expando option:
![image](https://user-images.githubusercontent.com/22506439/191857313-2fcec81b-42fa-47a2-9552-9bc6dc40d89e.png)

Video demo to see it in action;
https://user-images.githubusercontent.com/22506439/191857353-ee5bff89-7aef-4488-882c-73b3de544c99.mp4

I see there is an existing old implementation in #5358, which may still work.  It however uses `https://opengraph.io/` and requires an `app_id` query parameter to extract the embed link, in contrast, the implementation in this PR uses `https://web.scraper.workers.dev/` to do the same thing, just without any required authentication.  

With that being said, it is a third-party dependency and could stop working or change it's underlying functionality, it may also end up hitting issues given YouTube's underlying implementation changes, although it is working at this time of raising this PR.

Relevant issue: #5344 
Other implementations: #5358 
Tested in browser: Chrome, Firefox

corylulu commented:
This API isn't doing anything other than being a proxy and returning the entire body of the uncached HTML body and searching for the link in the body. You could just as easily call Youtube directly. 

It's also easier to find in the `og:video:url` metadata tag. The reason [embed.ly](https://github.com/reddit-archive/reddit/blob/3db4066f73753546bfb3cdcf8ab06f05443b7636/r2/r2/lib/media.py) was used was because it cached the data so it wouldn't spam calls or have taxing responses. 

I have yet to find a better solution using the APIs so far, however.

thomasasfk commented:
> https://jsonlink.io/

Hey, good point.  With caching in mind, I forked the original worker repository and changed it to just specifically handle our use-case with caching.  

You can see the project here: https://github.com/thomasasfk/web.scraper.workers.dev
And I've added changed to support it in:

jonas1015119 commented:
is there a reason why this isn't being added?

creesch commented:
> is there a reason why this isn't being added?

No specific reason that this isn't specifically added, but in general: 
https://www.reddit.com/r/RESAnnouncements/comments/sh83gx/announcement_life_of_reddit_enhancement_suite/

jonas1015119 commented:
ah, thanks. I guess this is isn't something that could just be added via Tampermonkey by users?
